tests/ClientTest.php

```diff
@@ -135,6 +135,43 @@ public function testCapture(): void
         $this->assertEquals('500a339f3ab2450b96dee542adf36ba7', $client->captureEvent($inputData));
     }
 
+    /**
+     * @dataProvider captureEventAttachesStacktraceAccordingToAttachStacktraceOptionDataProvider
+     */
+    public function testCaptureEventAttachesStacktraceAccordingToAttachStacktraceOption(bool $shouldAttachStacktrace): void
+    {
+        /** @var TransportInterface&MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
+        $transport->expects($this->once())
+            ->method('send')
+            ->with($this->callback(static function (Event $event) use ($shouldAttachStacktrace): bool {
+                if ($shouldAttachStacktrace && null === $event->getStacktrace()) {
```

Comment:
Couldn't you simplify this with a `return $shouldAttachStacktrace && null !== $event->getStacktrace();`?

Reply:
No, when `$shouldAttachStacktrace` is `false` then the stacktrace must not be present, and that `if` statement would compute as `false && false => false` when instead it should return `true`